woocommerce/woocommerce-ios
# Background image upload: show failure notice from saving product images

Part of #7021 
- [x] ⚠️ Please make sure https://github.com/woocommerce/woocommerce-ios/pull/7145 is approved before reviewing ⚠️ 



### Description


As suggested from showing a notice from image upload failure in https://github.com/woocommerce/woocommerce-ios/pull/7145, it'd be helpful to also show a notice from an error updating product images. In this PR, a new error enum `ProductImageUploaderError` was created for these two error scenarios (action handler like media upload API request, and saving product images at the end). `ProductImageUploadError` was also renamed to `ProductImageUploadErrorInfo` to minimize confusion.

### Testing instructions


One way to test image upload failure is by updating the [product images remote endpoint](https://github.com/woocommerce/woocommerce-ios/blob/3c2e5f2f08da760099517423061957d21b31a5dc/Networking/Networking/Remote/ProductsRemote.swift#L287) to an invalid one like adding some extra character at the end. You can also put `sleep(numberOfSeconds)` to simulate a longer response time.

- Launch the app
- In the product form, add one or more images
- Tap "Save" to save the product while the images are still being uploaded
- Leave the product form quickly, and feel free to do anything in the app including switching stores --> when the image(s) is uploaded but the product's images cannot be saved, an in-app notice (snackbar) should be shown with a title `Error saving product images` and an action `View` (if you are viewing a modal when the failure occurs, the notice is shown after dismissing the modal)
- Tap on "View" to view product details --> it should switch to the original store if needed, and present the product form. There should not be a "Save" CTA in the product form

### Screenshots



https://user-images.githubusercontent.com/1945542/176068841-4c535c80-080e-4c62-b817-b43e09527d6c.mov



---
